include/capo/pcm.hpp

```diff
@@ -12,13 +12,17 @@ using SampleRate = std::size_t;
 struct SampleMeta {
 	SampleRate rate{};
 	SampleFormat format{};
-	std::size_t channels{};
+	std::size_t totalFrameCount{};
+	Time length{};
 	static constexpr std::size_t max_channels_v = 2;
 
-	constexpr std::size_t sampleCount(std::size_t pcmFrameCount) noexcept { return pcmFrameCount * channels; }
-	constexpr bool supported() noexcept {
-		if (channels == 0 || channels > max_channels_v) { return false; }
-		return rate > 0;
+	static constexpr bool supported(std::size_t channels) noexcept { return channels > 0 && channels <= max_channels_v; }
```

Comment:
Why did we move away from the old `supported()`?

Reply:
Since it is now impossible to represent > 2 channels in `SampleMeta`, the old way was basically redundant.